plugins/inventory/aws_mq.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+name: aws_mq
+short_description: MQ broker inventory source
+description:
+  - Get brokers from Amazon Web Services MQ.
+  - Uses a YAML configuration file that ends with aws_mq.(yml|yaml).
+options:
+  regions:
+    description:
+      - A list of regions in which to describe MQ brokers. Available regions are listed here
+        U(https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/)
+    type: list
+    elements: str
+    default: []
+  strict_permissions:
+    description: By default if an AccessDenied exception is encountered this plugin will fail. You can set strict_permissions to
+      False in the inventory config file which will allow the restrictions to be gracefully skipped.
+    type: bool
+    default: True
+  statuses:
+    description:
+      - A list of desired states for brokers to be added to inventory. Set to ['all'] as a shorthand to find everything.
+        Possible value are listed here U(https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/broker-statuses.html)
+    type: list
+    elements: str
+    default:
+      - RUNNING
+      - CREATION_IN_PROGRESS
+  hostvars_prefix:
+    description:
+      - The prefix for host variables names coming from AWS.
+    type: str
+  hostvars_suffix:
+    description:
+      - The suffix for host variables names coming from AWS.
+    type: str
+extends_documentation_fragment:
+  - inventory_cache
+  - constructed
+  - amazon.aws.boto3
+  - amazon.aws.common.plugins
+  - amazon.aws.region.plugins
+  - amazon.aws.assume_role.plugins
+author:
+  - Ali AlKhalidi (@doteast)
+"""
+
+EXAMPLES = r"""
+plugin: community.aws.aws_mq
```

Comment:
It would be great if you could add more examples in here.

Reply:
added